data/src/main/java/org/akvo/flow/data/net/FlowRestApi.java

```diff
@@ -20,12 +20,137 @@
 
 package org.akvo.flow.data.net;
 
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+import android.util.Base64;
+
+import org.akvo.flow.data.entity.ApiLocaleResult;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
 import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import rx.Observable;
+import timber.log.Timber;
 
+@Singleton
 public class FlowRestApi {
```

Comment:
Similar question here.  Do you see this as the definition class for all API requests or only syncing data points?